services/121-service/src/transaction-job-processors/transaction-job-processors.service.ts

```diff
@@ -0,0 +1,334 @@
+import { getScopedRepositoryProviderName } from '@121-service/src/utils/scope/createScopedRepositoryProvider.helper';
+import { Inject, Injectable } from '@nestjs/common';
+
+@Injectable()
+export class TransactionJobProcessorsService {
+  public constructor(
+    private readonly intersolveVisaService: IntersolveVisaService,
+    private readonly messageTemplateService: MessageTemplateService,
+    private readonly programFinancialServiceProviderConfigurationRepository: ProgramFinancialServiceProviderConfigurationRepository,
+    private readonly registrationScopedRepository: RegistrationScopedRepository,
+    private readonly queueMessageService: QueueMessageService,
+    @Inject(getScopedRepositoryProviderName(TransactionEntity))
+    private readonly transactionScopedRepository: ScopedRepository<TransactionEntity>,
+    private readonly financialServiceProviderRepository: FinancialServiceProviderRepository,
+    private readonly latestTransactionRepository: LatestTransactionRepository,
+    private readonly programRepository: ProgramRepository,
+    private readonly eventsService: EventsService,
+  ) {}
+
+  public async processIntersolveVisaTransactionJob(
+    input: IntersolveVisaTransactionJobDto,
+  ): Promise<void> {
+    const registration =
+      await this.registrationScopedRepository.getByReferenceId({
+        referenceId: input.referenceId,
+      });
+    if (!registration) {
+      throw new Error(
+        `Registration was not found for referenceId ${input.referenceId}`,
+      );
+    }
+    const oldRegistration = structuredClone(registration);
+    const financialServiceProvider =
+      await this.financialServiceProviderRepository.getByName(
+        FinancialServiceProviderName.intersolveVisa,
+      );
+    if (!financialServiceProvider) {
+      throw new Error('Financial Service Provider not found');
+    }
+
+    // Check if all required properties are present. If not, create a failed transaction and throw an error.
+    for (const key in input) {
+      if (key === 'addressHouseNumberAddition') continue; // Skip non-required property
+
+      // Define "empty" based on your needs. Here, we check for null, undefined, or an empty string.
+      if (
+        input[key] === null ||
+        input[key] === undefined ||
+        input[key] === ''
+      ) {
+        const errorText = `Property ${key} is undefined`;
+        await this.createTransaction({
+          amount: input.transactionAmount,
+          registration: registration,
+          financialServiceProviderId: financialServiceProvider.id,
+          programId: input.programId,
+          paymentNumber: input.paymentNumber,
+          userId: input.userId,
+          status: StatusEnum.error,
+          errorMessage: errorText,
+        });
+        throw new Error(errorText);
+      }
+    }
+    const intersolveVisaConfig =
+      await this.programFinancialServiceProviderConfigurationRepository.getValuesByNamesOrThrow(
```

Comment:
Good point. 
I think that (2) may not be so difficult. So that could be preferred: the payment fails upon start, with a descriptive error to the user (think about which HTTP status code). And then CVA IM first has to correct the FSP config and then let the user try again.

Reply:
Opt for option 2 - try to make a generic check for all FSP to see if required config is available